gordo_components/builder/build_model.py

```diff
@@ -32,593 +33,577 @@
+        self.metadata = metadata.copy()
+
+    def build(self) -> Tuple[Optional[sklearn.base.BaseEstimator], dict]:
+        """
+        Build the model using the current state of the Builder
+
+        Returns
+        -------
+            Tuple[Optional[sklearn.base.BaseEstimator], dict]
+        """
+        if self.evaluation_config.get("seed"):
+            self.set_seed(seed=self.evaluation_config["seed"])
+
+        # Get the dataset from config
+        logger.debug(f"Initializing Dataset with config {self.data_config}")
+
+        dataset = (
+            self.data_config
+            if isinstance(self.data_config, GordoBaseDataset)
+            else _get_dataset(self.data_config)
+        )
 
-    # Get the dataset from config
-    logger.debug(f"Initializing Dataset with config {data_config}")
+        logger.debug("Fetching training data")
+        start = time.time()
 
-    dataset = (
-        data_config
-        if isinstance(data_config, GordoBaseDataset)
-        else _get_dataset(data_config)
-    )
+        X, y = dataset.get_data()
 
-    logger.debug("Fetching training data")
-    start = time.time()
+        time_elapsed_data = time.time() - start
 
-    X, y = dataset.get_data()
+        # Get the model and dataset
+        logger.debug(f"Initializing Model with config: {self.model_config}")
+        model = serializer.pipeline_from_definition(self.model_config)
 
-    time_elapsed_data = time.time() - start
+        cv_duration_sec = None
 
-    # Get the model and dataset
-    logger.debug(f"Initializing Model with config: {model_config}")
-    model = serializer.pipeline_from_definition(model_config)
+        scores: Dict[str, Any] = dict()
+        if self.evaluation_config["cv_mode"].lower() in (
+            "cross_val_only",
+            "full_build",
+        ):
 
-    cv_duration_sec = None
+            # Build up a metrics list.
+            metrics_list = self.metrics_from_list(self.evaluation_config.get("metrics"))
 
-    scores: Dict[str, Any] = dict()
-    if evaluation_config["cv_mode"].lower() in ("cross_val_only", "full_build"):
+            # Cross validate
+            logger.debug("Starting cross validation")
+            start = time.time()
+            if hasattr(model, "predict"):
 
-        # Build up a metrics list.
-        metrics_list = metrics_from_list(evaluation_config.get("metrics"))
+                scaler = self.evaluation_config.get("scoring_scaler")
+                metrics_dict = self.build_metrics_dict(metrics_list, y, scaler=scaler)
 
-        # Cross validate
-        logger.debug("Starting cross validation")
-        start = time.time()
-        if hasattr(model, "predict"):
+                cv = cross_validate(
+                    model,
+                    X,
+                    y,
+                    scoring=metrics_dict,
+                    return_estimator=True,
+                    cv=TimeSeriesSplit(n_splits=3),
```

Comment:
Possbile to specify a different number of splits?

Reply:
Oh nice. :+1:

Reply:
I started, but I think maybe it's better to leave this until after this is merged. So far as I can see, maybe this should be tackled with the whole of the ability to define the splitter in the `evaluation` key. 

ie. 
```yaml
evaluation:
  cv:
    sklearn.model_selection.TimeSeriesSplit:
      n_splits: 5
```

Reply:
I agree, I think you have mentioned this before and it sounds like a good idea.